src/views/Header.js

```diff
@@ -113,9 +113,17 @@ class Header extends React.PureComponent<void, HeaderProps, HeaderState> {
 
   _renderLeftComponent = (props: SceneProps) => {
     const options = this.props.getScreenDetails(props.scene).options;
-    if (typeof options.headerLeft !== 'undefined') {
-      return options.headerLeft;
+
+    const { headerLeft } = options;
+
+    if (typeof headerLeft !== 'undefined' && typeof headerLeft !== 'function') {
+      return headerLeft;
     }
+
+    if (typeof headerLeft === 'function') {
+      return headerLeft(this.props);
```

Comment:
how about adding the `scene` to the arguments, like:
```
return headerLeft(this.props, props.scene)
```
That will let the button component, for example, return null if it's the only screen (see check on line 127)